0.20.1  Still getting error on migration from h2 to mysql, PermissionsGroup Duplicate Entry

metabase 0.20.1 on an ubuntu aws instance

Is there a set of commands I can run to clear this?

`Transfering 3 instances of PermissionsGroup....BatchUpdateException:
 Message: Duplicate entry '3' for key 'PRIMARY'
 SQLState: 23000
 Error Code: 1062
java.sql.BatchUpdateException: Duplicate entry '3' for key 'PRIMARY'
	at com.mysql.jdbc.SQLError.createBatchUpdateException(SQLError.java:1166)
	at com.mysql.jdbc.PreparedStatement.executeBatchSerially(PreparedStatement.java:1773)
	at com.mysql.jdbc.PreparedStatement.executeBatchInternal(PreparedStatement.java:1257)
	at com.mysql.jdbc.StatementImpl.executeBatch(StatementImpl.java:958)
	at clojure.java.jdbc$execute_batch.invokeStatic(jdbc.clj:426)
	at clojure.java.jdbc$execute_batch.invoke(jdbc.clj:419)
	at clojure.java.jdbc$db_do_execute_prepared_statement$fn__2749.invoke(jdbc.clj:792)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:637)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:598)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invokeStatic(jdbc.clj:791)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invoke(jdbc.clj:778)
	at clojure.java.jdbc$db_do_prepared.invokeStatic(jdbc.clj:814)
	at clojure.java.jdbc$db_do_prepared.invoke(jdbc.clj:795)
	at clojure.java.jdbc$insert_cols_BANG_.invokeStatic(jdbc.clj:1069)
	at clojure.java.jdbc$insert_cols_BANG_.invoke(jdbc.clj:1061)
	at clojure.java.jdbc$insert_multi_BANG_.invokeStatic(jdbc.clj:1110)
	at clojure.java.jdbc$insert_multi_BANG_.invoke(jdbc.clj:1093)
	at metabase.cmd.load_from_h2$insert_entity_BANG_$fn__31010.invoke(load_from_h2.clj:125)
	at metabase.cmd.load_from_h2$insert_entity_BANG_.invokeStatic(load_from_h2.clj:124)
	at metabase.cmd.load_from_h2$insert_entity_BANG_.invoke(load_from_h2.clj:106)
	at metabase.cmd.load_from_h2$load_data_BANG_.invokeStatic(load_from_h2.clj:140)
	at metabase.cmd.load_from_h2$load_data_BANG_.invoke(load_from_h2.clj:134)
	at metabase.cmd.load_from_h2$load_from_h2_BANG_$fn__31090.invoke(load_from_h2.clj:209)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:613)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:629)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:598)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at metabase.cmd.load_from_h2$load_from_h2_BANG_.invokeStatic(load_from_h2.clj:206)
	at metabase.cmd.load_from_h2$load_from_h2_BANG_.invoke(load_from_h2.clj:199)
	at clojure.lang.Var.invoke(Var.java:379)
	at metabase.core$load_from_h2.invokeStatic(core.clj:218)
	at metabase.core$load_from_h2.invoke(core.clj:211)
	at metabase.core$load_from_h2.invokeStatic(core.clj:214)
	at metabase.core$load_from_h2.invoke(core.clj:211)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:646)
	at clojure.core$apply.invoke(core.clj:641)
	at metabase.core$run_cmd$fn__31843.invoke(core.clj:265)
	at metabase.core$run_cmd.invokeStatic(core.clj:265)
	at metabase.core$run_cmd.doInvoke(core.clj:264)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.core$apply.invokeStatic(core.clj:648)
	at clojure.core$apply.invoke(core.clj:641)
	at metabase.core$_main.invokeStatic(core.clj:279)
	at metabase.core$_main.doInvoke(core.clj:275)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at metabase.core.main(Unknown Source)
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '3' for key 'PRIMARY'
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:404)
	at com.mysql.jdbc.Util.getInstance(Util.java:387)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:934)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3966)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3902)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2526)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2673)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2549)
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1861)
	at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2073)
	at com.mysql.jdbc.PreparedStatement.executeBatchSerially(PreparedStatement.java:1751)
	... 52 more
Command failed with exception: Duplicate entry '3' for key 'PRIMARY'`

--- a/src/metabase/db/migrations.clj
+++ b/src/metabase/db/migrations.clj
@@ -1,5 +1,11 @@
 (ns metabase.db.migrations
-  "Clojure-land data migration definitions and fns for running them."
+  "Clojure-land data migration definitions and fns for running them.
+   These migrations are all ran once when Metabase is first launched, except when transferring data from an existing H2 database.
+   When data is transferred from an H2 database, migrations will already have been ran against that data; thus, all of these migrations
+   need to be repeatable, e.g.:
+
+     CREATE TABLE IF NOT EXISTS ... -- Good
+     CREATE TABLE ...               -- Bad"
   (:require [clojure.string :as s]
             [clojure.tools.logging :as log]
             (metabase [config :as config]
@@ -228,13 +234,15 @@
         {admin-group-id :id}     (perm-group/admin)]
     (binding [perm-membership/*allow-changing-all-users-group-members* true]
       (doseq [{user-id :id, superuser? :is_superuser} (db/select [User :id :is_superuser])]
-        (db/insert! PermissionsGroupMembership
-          :user_id  user-id
-          :group_id all-users-group-id)
-        (when superuser?
+        (u/ignore-exceptions
           (db/insert! PermissionsGroupMembership
             :user_id  user-id
-            :group_id admin-group-id))))))
+            :group_id all-users-group-id))
+        (when superuser?
+          (u/ignore-exceptions
+            (db/insert! PermissionsGroupMembership
+              :user_id  user-id
+              :group_id admin-group-id)))))))
 
 ;; admin group has a single entry that lets it access to everything
 (defmigration add-admin-group-root-entry

